package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -78,14 +82,18 @@ def layers_mapping(self):
             if not metadata:
                 continue
             try:
-                dataset_layer = dataset.metadata["kedro-viz"]["layer"]
+                layer = dataset.metadata["kedro-viz"]["layer"]
             except (AttributeError, KeyError):  # pragma: no cover
                 logger.debug(
                     "No layer info provided under metadata in the catalog for %s",
                     dataset_name,
                 )
             else:
-                self._layers_mapping[self.strip_encoding(dataset_name)] = dataset_layer
+                if TRANSCODING_SEPARATOR in dataset_name:
```

Comment:
Same as above comment, we don't need to check for the transcoding separator here

Reply:
as discussed, we will keep the transcoding_seperator if statement for now so that validation doesn't happen for all datasets just the transcoding ones also because transcoding is not very common.